frontend_new/src/components/Cards.tsx

```diff
@@ -8,40 +8,74 @@ import Typography from "@material-ui/core/Typography";
 import Grid from "@material-ui/core/Grid";
 import { customTheme } from "../themes/customTheme";
 import ListItem from "@material-ui/core/ListItem";
+import BookmarkBorderOutlinedIcon from "@material-ui/icons/BookmarkBorderOutlined";
+import BookmarkOutlinedIcon from "@material-ui/icons/BookmarkOutlined";
+import IconButton from "@material-ui/core/IconButton";
+import ListItemSecondaryAction from "@material-ui/core/ListItemSecondaryAction";
+import Paper from "@material-ui/core/Paper";
 
 type props = {
   web_link: string;
   primary: string;
   secondary: string;
+  id: integer;
+  handlebookmark: () => void;
+  bookmarks: array;
+  isVisible: boolean;
 };
 function gotoLink(url: string) {
   window.location.href = url;
 }
 
 export default function SimpleCard(props: props) {
+  const [bookmarked, setBookmarked] = React.useState(true);
+  function handlebookmark() {
+    props.handlebookmark(
+      props.web_link,
+      props.primary,
+      props.secondary,
+      props.id
+    );
+    if (props.id in props.bookmarks) {
+      setBookmarked(true);
+    } else {
+      setBookmarked(false);
+    }
+  }
+
+  React.useEffect(() => {
+    console.log("in use effect");
```

Comment:
**[Minor, Required]** Just something I came across while going over all PRs, worth noting. We shouldn't be logging anything to console in any case other than local testing. Probably just remove this.

Reply:
removed it in latest commit!